internal/controller/policyserver_controller_deployment.go

```diff
@@ -220,28 +225,111 @@ func (r *PolicyServerReconciler) updatePolicyServerDeployment(policyServer *poli
+			policyServerDeployment.Spec.Template.Spec.Volumes = append(policyServerDeployment.Spec.Template.Spec.Volumes, corev1.Volume{
+				Name: otelCertificateVolumeName,
+				VolumeSource: corev1.VolumeSource{
+					Secret: &corev1.SecretVolumeSource{
+						SecretName: r.OtelCertificateSecret,
+					},
+				},
+			})
+			admissionContainer.VolumeMounts = append(admissionContainer.VolumeMounts, corev1.VolumeMount{
+				Name:      otelCertificateVolumeName,
+				ReadOnly:  true,
+				MountPath: otelCertificateVolumePath,
+			})
+			admissionContainer.Args = append(admissionContainer.Args, "--otlp-certificate="+otelCertificateVolumePath+"/ca.crt")
+		}
+		if len(r.OtelClientCertificateSecret) > 0 {
+			policyServerDeployment.Spec.Template.Spec.Volumes = append(policyServerDeployment.Spec.Template.Spec.Volumes, corev1.Volume{
+				Name: otelClientCertificateVolumeName,
+				VolumeSource: corev1.VolumeSource{
+					Secret: &corev1.SecretVolumeSource{
+						SecretName: r.OtelClientCertificateSecret,
+					},
+				},
+			})
+			admissionContainer.VolumeMounts = append(admissionContainer.VolumeMounts, corev1.VolumeMount{
+				Name:      otelClientCertificateVolumeName,
+				ReadOnly:  true,
+				MountPath: otelClientCertificateVolumePath,
+			})
+
+			admissionContainer.Args = append(admissionContainer.Args, "--otlp-client-certificate="+otelClientCertificateVolumePath+"/tls.crt")
+			admissionContainer.Args = append(admissionContainer.Args, "--otlp-client-certificate-key="+otelClientCertificateVolumePath+"/tls.key")
+		}
+		// As the controller is sending data to remote otel collector, we need
+		// to replicate the env vars to the policy server deployment. Thus, it
+		// will be able to send data to the same collector.
+		otelEnvVarToReplicate := []string{
+			"OTEL_EXPORTER_OTLP_CERTIFICATE",
+			"OTEL_EXPORTER_OTLP_CLIENT_CERTIFICATE",
+			"OTEL_EXPORTER_OTLP_CLIENT_KEY",
+			"OTEL_EXPORTER_OTLP_COMPRESSION",
+			"OTEL_EXPORTER_OTLP_HEADERS",
+			"OTEL_EXPORTER_OTLP_INSECURE",
+			"OTEL_EXPORTER_OTLP_METRICS_CERTIFICATE",
+			"OTEL_EXPORTER_OTLP_METRICS_CLIENT_CERTIFICATE",
+			"OTEL_EXPORTER_OTLP_METRICS_CLIENT_KEY",
+			"OTEL_EXPORTER_OTLP_METRICS_COMPRESSION",
+			"OTEL_EXPORTER_OTLP_METRICS_DEFAULT_HISTOGRAM_AGGREGATION",
+			"OTEL_EXPORTER_OTLP_METRICS_ENDPOINT",
+			"OTEL_EXPORTER_OTLP_METRICS_HEADERS",
+			"OTEL_EXPORTER_OTLP_METRICS_INSECURE",
+			"OTEL_EXPORTER_OTLP_METRICS_TEMPORALITY_PREFERENCE",
+			"OTEL_EXPORTER_OTLP_METRICS_TIMEOUT",
+			"OTEL_EXPORTER_OTLP_TIMEOUT",
+			"OTEL_EXPORTER_OTLP_TRACES_CERTIFICATE",
+			"OTEL_EXPORTER_OTLP_TRACES_CLIENT_CERTIFICATE",
+			"OTEL_EXPORTER_OTLP_TRACES_CLIENT_KEY",
+			"OTEL_EXPORTER_OTLP_TRACES_COMPRESSION",
+			"OTEL_EXPORTER_OTLP_TRACES_ENDPOINT",
+			"OTEL_EXPORTER_OTLP_TRACES_HEADERS",
+			"OTEL_EXPORTER_OTLP_TRACES_INSECURE",
+			"OTEL_EXPORTER_OTLP_TRACES_TIMEOUT",
+		}
+		// "OTEL_EXPORTER_OTLP_ENDPOINT" is not replicated because the endpoint is passed using CLI flag.
```

Comment:
Can't we add the code on the policy server side to handle this as an env variable too? Just for the sake of consistency and to keep this code simpler (not have to maintain the cognitive overload of what is to be passed as flag vs cli flag).

CC @fabriziosestito

Reply:
> Can't we add the code on the policy server side to handle this as an env variable too

I think this is the way to go. I misunderstood, as I thought we were exposing only the endpoint and the certificate settings. 

To summarize, we should only use environment variables.
 I am updating the policy server to mimic the missing upstream logic for the certificates env vars.